src/map.cpp

```diff
@@ -137,4 +138,93 @@ Map::~Map()
     close(mapfd_);
 }
 
+void MapManager::Add(std::unique_ptr<IMap> map)
+{
+  auto name = map->name_;
+  if (name.empty())
+    throw std::runtime_error("Map without name cannot be stored");
```

Comment:
I think it'd be nicer to return a bool from this function instead of throwing an exception

Reply:
This is here to indicate a bug. All user maps should be named, for internal maps Set() should be used. I think that exceptions are the most appropriate thing we have atm for bugs as you want to keep the message somewhere.

Reply:
Sure, makes sense